internal/data/embedded_wallet.go

```
@@ -0,0 +1,172 @@
+package data
+
+import (
+	"context"
+	"database/sql"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/stellar/go/strkey"
+
+	"github.com/stellar/stellar-disbursement-platform-backend/db"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/utils"
+)
+
+type EmbeddedWalletStatus string
+
+const (
+	// The token has been created but the wallet has not been created yet.
+	PendingWalletStatus EmbeddedWalletStatus = "PENDING"
+	// The wallet creation is in progress.
+	ProcessingWalletStatus EmbeddedWalletStatus = "PROCESSING"
+	// The wallet has been created and is ready to use.
+	SuccessWalletStatus EmbeddedWalletStatus = "SUCCESS"
+	// The wallet creation failed.
+	FailedWalletStatus EmbeddedWalletStatus = "FAILED"
+)
+
+func (status EmbeddedWalletStatus) Validate() error {
+	switch EmbeddedWalletStatus(strings.ToUpper(string(status))) {
+	case PendingWalletStatus, SuccessWalletStatus, ProcessingWalletStatus, FailedWalletStatus:
+		return nil
+	default:
+		return fmt.Errorf("invalid embedded wallet status %q", status)
+	}
+}
```

Are these statuses equivalent to the RPC ones? If so, I suggest renaming `EmbeddedWalletStatus` to something more RPC-relatable

---

Not really, this is closer to the payment status. Any RPC/Horizon errors are hidden away in the TSS database.